components/tenant-mgt/org.wso2.carbon.tenant.mgt/src/main/java/org/wso2/carbon/tenant/mgt/services/TenantMgtAdminService.java

```diff
@@ -154,6 +154,17 @@ private void notifyTenantAddition(TenantInfoBean tenantInfoBean) throws Exceptio
         }
     }
 
+    private void notifyPreTenantAddition(TenantInfoBean tenantInfoBean) throws Exception {
+
+        try {
+            TenantMgtUtil.triggerPreAddTenant(tenantInfoBean);
+        } catch (StratosException e) {
+            String msg = "Error in notifying pre tenant addition.";
+            log.error(msg, e);
```

Comment:
Do we need to log and throw here?

Reply:
exception gets tansformed. And this is the final place the exception is handled in the service.

Reply:
But you add the current exception when you are wrapping it. So ideally it should be propagated to the next level.

Reply:
Since this is a admin service this will be the last level in IS